src/model.jl

```diff
@@ -490,48 +578,176 @@ const perihelion_parameters_phase = [251.9025, 280.8325, 128.3057, 348.1074, 292
-function perihelion(year)
+  end
+  
+  
+  # Eqn. 6 from Berger 1978, Long-Term Variations of daily Insolation and Quaternary Climatic Changes
+  function perihelion(year)
   F     = perihelion_parameters_amplitude
   f     = perihelion_parameters_mean_rate
   delta = perihelion_parameters_phase
-
+  
   pi_ = atan(e_sincos_pi(year)...)
   psi_tilde = 50.439273
   zeta = 3.392506
-
+  
   psi = psi_tilde/3600 * (year-1950) + zeta + sum(F/3600 .* sin.(deg2rad.(f/3600) * (year-1950) .+ deg2rad.(delta)))
-
+  
   return (pi_ + deg2rad(psi)) % (2*pi)
+  end
+
+
+
+# possible years to pick annual sea ice extent for the northern hemisphere : 1979 to 2020
+# returns a vector with sea ice extent per month from March to February
+
+function read_sea_ice_extent_N(year=1979)
+
+  matrix = readdlm(joinpath(@__DIR__,"..","input/sea_ice_data/","N_ALL_extent_v3.0.dat"),comments=true)
+  yearindex = findfirst(isequal(year),matrix[:,1])
+  annual_sea_ice_extent = matrix[yearindex,2:13]
+
+  return annual_sea_ice_extent
 end
+
+
+
+"""
+calc_geography_per_month_extent()
+Determines a new geography matrix by determining sea ice extent with a monthly rate of change
+and mapping extent or reduction to adjacent sea ice cells and ocean cells on the Northern Hemisphere.
+
+"""
+
+function calc_geography_per_month_extent(geography,annual_sea_ice_extent, time_step, nlongitude=128, nlatitude =65)
+
+  month::Int64 = 0
+
+ # Gets the right month when the simulation starts at the vernal equinox
+ if time_step in (1,46,47,48)
+  month = 1
+  elseif mod(time_step,4) == 2
+    month = 1
+    for nt in 1:time_step
+      if mod(nt,4) == 1
+        month +=1
+      end
+    end   
+ end
+
+  # Gets the right month when the simulation starts with the first week of January 
+  #for nt in 1:time_step
+  # if mod(nt,4) == 1
+  # month +=1
+  # end
+  #end
+  
+  nlat::Int64 = (nlatitude-1)*0.5
+  geography_start = read_geography(joinpath(@__DIR__, "..", "input", "The_World.dat"),128,65)
+  geo = geography_start[:,1:nlat]
+  sea_ice_index_unit = size(findall(isequal(2),geo),1)/annual_sea_ice_extent[1]
+  
+  
+
+  if month == 0 || month == 1
+    geography = geography_start
+    return geography
+  else
+  
+   # collect submatrix and indices for the Northern Hemisphere
+
+   G = geography[:,1:nlat]
+   N = reverse(G, dims = 2)
+   IndG = CartesianIndices(G)
+   IndN = CartesianIndices(N)
+   Gfirst, Glast = first(IndG), last(IndG)
+   Ifirst, Ilast = first(IndN), last(IndN)
+   G1 = oneunit(Gfirst)
+   I1 = oneunit(Ifirst)
+
+
+   # get all sea ice cells for the Northern Hemispehre and determine how many cells will be affected by the extent/reduction
+
+   sea_ice_index = findall(isequal(2),G)
+   old_counter = size(sea_ice_index,1)
+   new_counter = sea_ice_index_unit*annual_sea_ice_extent[month]
+   new_cells = abs(floor(Int,new_counter - old_counter))
+
+
+   # case for an extent of sea ice
+   # adjacent ocean cells will turn to sea ice cells
+
```

Comment:
```suggestion
if new_cells == 0
  return geo
end
```
You can consider the case when you don't need to change the number of sea ice cells here. Please update when you change the name of variable `new_cells`